<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 2 Genetic Ancestry | Statistical Genetics Content Summary</title>
  <meta name="description" content=" 2 Genetic Ancestry | Statistical Genetics Content Summary" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content=" 2 Genetic Ancestry | Statistical Genetics Content Summary" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 2 Genetic Ancestry | Statistical Genetics Content Summary" />
  
  
  

<meta name="author" content="Alayna Johnson" />


<meta name="date" content="2025-03-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genome-wide-association-studies-gwas.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Content Summary</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="1" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html"><i class="fa fa-check"></i><b>1</b> Genome-Wide Association Studies (GWAS)</a>
<ul>
<li class="chapter" data-level="1.1" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#introduction-to-gwas"><i class="fa fa-check"></i><b>1.1</b> Introduction to GWAS</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#genome-breakdown"><i class="fa fa-check"></i><b>1.1.1</b> Genome Breakdown</a></li>
<li class="chapter" data-level="1.1.2" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#a-small-example"><i class="fa fa-check"></i><b>1.1.2</b> A small example</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#simulating-a-gwas"><i class="fa fa-check"></i><b>1.2</b> Simulating a GWAS</a></li>
<li class="chapter" data-level="1.3" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#marginal-regression"><i class="fa fa-check"></i><b>1.3</b> Marginal Regression</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#multiple-testing"><i class="fa fa-check"></i><b>1.3.1</b> Multiple testing</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#visualizing-gwas"><i class="fa fa-check"></i><b>1.4</b> Visualizing GWAS</a></li>
<li class="chapter" data-level="" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#wrapping-up"><i class="fa fa-check"></i>Wrapping up</a></li>
<li class="chapter" data-level="" data-path="genome-wide-association-studies-gwas.html"><a href="genome-wide-association-studies-gwas.html#glossary"><i class="fa fa-check"></i>Glossary</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="genetic-ancestry.html"><a href="genetic-ancestry.html"><i class="fa fa-check"></i><b>2</b> Genetic Ancestry</a>
<ul>
<li class="chapter" data-level="2.1" data-path="genetic-ancestry.html"><a href="genetic-ancestry.html#describing-human-populations"><i class="fa fa-check"></i><b>2.1</b> Describing Human Populations</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="genetic-ancestry.html"><a href="genetic-ancestry.html#population-admixture"><i class="fa fa-check"></i><b>2.1.1</b> Population Admixture</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="genetic-ancestry.html"><a href="genetic-ancestry.html#principal-component-analysis"><i class="fa fa-check"></i><b>2.2</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="2.3" data-path="genetic-ancestry.html"><a href="genetic-ancestry.html#confounding-variables"><i class="fa fa-check"></i><b>2.3</b> Confounding Variables</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="genetic-ancestry.html"><a href="genetic-ancestry.html#omitted-variable-bias"><i class="fa fa-check"></i><b>2.3.1</b> Omitted Variable Bias</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="genetic-ancestry.html"><a href="genetic-ancestry.html#glossary-1"><i class="fa fa-check"></i>Glossary</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Genetics Content Summary</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genetic-ancestry" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number"> 2</span> Genetic Ancestry<a href="genetic-ancestry.html#genetic-ancestry" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>What is ancestry, and what can genetic data tell us? There are several ways that one can think about genetic ancestry. In these chapters, we will consider genetic ancestry to loosely mean the genetic relationship between you and your ancestors.</p>
<p>An important idea to remember throughout this chapter is <em>genetic similarity</em>. If you have ever taken a genetic test through a company like 23andMe or Ancestry.com you are most likely familiar with this concept. When you take one of these tests by sending in some of your DNA, these companies are reporting back to you how similar your DNA is compared to different populations.</p>
<p>The concept of different populations comes up a lot in these contexts. As with all scientific research, there are some ethical issues surrounding how researchers classify populations in genetic studies. We will get into this next.</p>
<p><br />
</p>
<div id="describing-human-populations" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Describing Human Populations<a href="genetic-ancestry.html#describing-human-populations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>When a researcher is working with genetic data, they will likely use some sort of population descriptor to attempt to capture complex differences in human genetics.</p>
<p>When you look at the results of your personal DNA test, you will likely see your ancestry broken down into smaller percentages of regions or peoples that contribute to 100% you. The way these regions are described brings with it some questions you may have never thought of before:</p>
<ul>
<li><p><em>How are regions defined? Are political borders used in every case?</em></p></li>
<li><p><em>What is the genetic “line” between being reported as native to one area versus another geography close by? How specific can this get?</em></p></li>
</ul>
<div id="population-admixture" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Population Admixture<a href="genetic-ancestry.html#population-admixture" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As our world has become so interconnected between geographies and people’s from long ago to the present, many of us have DNA inherited from multiple populations. The general term for this is <em>population admixture</em>.</p>
<p><br />
<br />
</p>
</div>
</div>
<div id="principal-component-analysis" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Principal Component Analysis<a href="genetic-ancestry.html#principal-component-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you have taken a class in linear algebra or machine learning, you may be familiar with the method of Principal Component Analysis (PCA). We use PCA as a technique to reduce the dimensionality of correlated variables while keeping as much variation as possible. We will be left with principal components (PC’s) as weighted combinations of these variables that explain the most variation in our data. Essentially, instead of having thousands of SNPs, this method will represent genetic variation in fewer dimensions made up of the original data.</p>
<p>The first few PC’s explain the most variation in our data. Here is an example of what the PC’s we create from this method look like:</p>
<p><span class="math display">\[\begin{align}
\text{PC}1 &amp;= a{11}x_1 + a_{12}x_2 + \dots + a_{1p}x_p \\
\text{PC}2 &amp;= a{21}x_1 + a_{22}x_2 + \dots + a_{2p}x_p \\
&amp;\vdots \\
\text{PC}n &amp;= a{n1}x_1 + a_{n2}x_2 + \dots + a_{np}x_p
\end{align}\]</span></p>
<p>where</p>
<p><span class="math display">\[\begin{align}
x_1, x_2, \dots, x_p \text{ are the original SNPs and} \\
a_{ij} \text{ are the weights of each SNP towards a PC}
\end{align}\]</span></p>
<p><br />
<br />
</p>
</div>
<div id="confounding-variables" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Confounding Variables<a href="genetic-ancestry.html#confounding-variables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="omitted-variable-bias" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Omitted Variable Bias<a href="genetic-ancestry.html#omitted-variable-bias" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><br />
</p>
</div>
</div>
<div id="glossary-1" class="section level2 unnumbered hasAnchor">
<h2>Glossary<a href="genetic-ancestry.html#glossary-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Terms for this chapter and beyond.</p>
<ul>
<li><p><u><em>Pedigree</em></u>: how genealogical ancestors are related to each other, like an extensive family tree</p></li>
<li><p><u><em>Genealogical ancestry</em></u>: the identifiable ancestors in your pedigree</p></li>
<li><p><u><em>Genetic ancestry</em></u>: the paths through one’s pedigree by which genetic traits have been inherited</p></li>
<li><p><u><em>Genetic similarity</em></u>: classifying genomes as similar to specific populations</p>
<ul>
<li><u><em>Population admixture</em></u>: the mating of individuals with different geographical and continental origins</li>
</ul></li>
<li><p><u><em>Principal Component Analysis</em></u>:</p></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genome-wide-association-studies-gwas.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/laynajohnson/statistical-genetics/edit/main/02-unit2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": "https://github.com/laynajohnson/statistical-genetics/blob/main/02-unit2.Rmd",
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"sharing": {
"linkedin": true
},
"search": true,
"search_options": {
"placeholder": "Search book..."
}
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
